libraries/rust/data/src/client.rs

```diff
@@ -0,0 +1,218 @@
+use chrono::{DateTime, Utc};
+use reqwest::Url;
+use reqwest::header::{HeaderMap, HeaderValue, ACCEPT};
+use reqwest::Client as HTTPClient;
+use aws_sdk_s3::client::Client as S3Client;
+use aws_credential_types::Credentials;
+use aws_types::region::Region;
+use aws_sdk_s3::Config;
+use serde::{Serialize, Deserialize};
+use serde_json;
+use aws_sdk_s3::primitives::ByteStream;
+use flate2::write::GzEncoder;
+use flate2::read::GzDecoder;
+use flate2::Compression;
+use std::io::{Write, Read};
+
+#[derive(Deserialize)]
+struct BarsResponse {
+    bars: Vec<Bar>,
+    #[serde(rename = "next_page_token")]
+    next_token: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Bar {
+    pub ticker: Option<String>,
+    #[serde(rename = "t")]
+    pub timestamp: DateTime<Utc>,
+    #[serde(rename = "o")]
+    pub open: f32,
+    #[serde(rename = "h")]
+    pub high: f32,
+    #[serde(rename = "l")]
+    pub low: f32,
+    #[serde(rename = "c")]
+    pub close: f32,
+    #[serde(rename = "v")]
+    pub volume: u32,
+    #[serde(rename = "n")]
+    pub number_of_trades: u32,
+    #[serde(rename = "vw")]
+    pub volume_weighted_average_price: f32,
+}
+
+pub struct Client {
+    alpaca_base_url: String,
+    alpaca_api_key_id: String,
+    alpaca_api_secret_key: String,
+    http_client: HTTPClient,
+    s3_client: S3Client,
+    s3_data_bucket_name: String,
+}
+
+const EQUITY_BARS_PATH: &str = "equity/bars";
+
+impl Client {
+    pub fn new(
+        alpaca_api_key_id: String,
+        alpaca_api_secret_key: String,
+        aws_access_key_id: String,
+        aws_secret_access_key: String,
+        s3_data_bucket_name: String,
+    ) -> Self {
+        let creds = Credentials::from_keys(aws_access_key_id, aws_secret_access_key, None);
+        
+        let conf = Config::builder()
+            .credentials_provider(creds)
+            .region(Region::new("us-east-1"))
+            .build();
+        
+        let s3_client = S3Client::from_conf(conf);
+
+        Client {
+            alpaca_base_url: "https://data.alpaca.markets/v2/".to_string(),
+            alpaca_api_key_id,
+            alpaca_api_secret_key,
+            http_client: HTTPClient::new(),
+            s3_client,
+            s3_data_bucket_name,
+        }
+    }
+
+    pub async fn fetch_equities_bars_by_date_range(
```

Comment:
nitpick but I feel like `Bar::fetch` is a sufficient name since date range is implied by the signature?

I dunno, in python this would be fine, but something about Rust it _feels_ like shorter?

feel free to ignore me, this is purely vibes-based

Reply:
Broadly agree but because it's a general `Bar` type being returned (not specifically equities) I'll still include the `equities_bars` part of the name.